calicoctl/commands/resources.go

```diff
@@ -124,6 +124,16 @@ func getResourceFromArguments(args map[string]interface{}) (unversioned.Resource
 			p.Metadata.CIDR = *cidr
 		}
 		return *p, nil
+	case "bgpPeer":
+		p := api.NewBGPPeer()
+		if name != "" {
+			err := p.Metadata.PeerIP.UnmarshalText([]byte(name))
+			if err != nil {
+				return nil, err
+			}
+		}
+		p.Metadata.Hostname = hostname
```

Comment:
Does this need to set `p.Metadata.Name` here as well?  Doesn't look like `name` is being used.

Also, so I understand, why is the defaulting of `Name` occurring here and not in libcalico?